### DISCUSSION TITLE
David Khourshid - xstate, stately.ai

### DISCUSSION THREAD

**Discussion by hipstersmoothie** [devtools-fm/devtools.fm#111] [Announcements]
## Background

[Twitter](https://twitter.com/DavidKPiano)
[GitHub](https://github.com/davidkpiano)
[Stately](https://stately.ai)
[xstate](https://xstate.js.org)

## Themes

- state machines

## Intro

Andrew: Hello, welcome to the devtools.fm podcast. This is a podcast about developer tools and the people who make them. I'm Andrew and this is my co-host Justin.

Justin: Hey everyone! I'm excited to welcome David Khourshid, the founder of Stately and the creator of XState. David, thanks for joining. I've wanted to talk about XState on the podcast for a while so this'll be a fun episode. Before we dig in, would you like to tell our listeners a little more about yourself?

### XState, State Charts, and State Machines

- [x] XState is a library for building finite state machines and state charts. Could you explain those terms for our listeners who may be unfamiliar?
- [x] What was your "aha" moment that led you to approaching code from a state machine perspective?
- [x] What types of code lend themselves well to state machines?
- [x] How might a state machine simplify my code?
- [x] What are common rough spots people hit when first trying to switch over to writing state machines?
- [x] What is the actor model and how does it relate to state machines?
- [x] Why did you build xstate? Were there alternatives at the time?
- [x] Where is the best place for a state machine? Do you use them on both the front end and back end? Share between?

### Stately

- [x] What is stately?
- [x] How does it relate to xstate?
- [x] Does using stately mean you rarely have to manually write out your state machines?
- [x] How do you pull it into your code?
- [x] Stately has collaboration built in, is the expectation that anyone on the team can help write the state machine?
- [x] Does it produce code that only works with react? or is it more of a general utility?
- [x] What's the ideal workflow using stately?
- [x] Where do you plan to take stately? What the eventual goal?
- [x] How do you plan to monetize this?

### Future

- [x] What do you think the future of state management is?

**Comment by hipstersmoothie** [comment]
My tooltips:

- https://preactjs.com/blog/introducing-signals/
- https://lightningcss.dev

**Comment by just-be-dev** [comment]
Tooltips:
- [Tweet on environment variable typing with Zod](https://twitter.com/mohsen____/status/1566837054444650499)
- [Dune, a hobby rust based JS runtime](https://github.com/aalykiot/dune)

**Comment by just-be-dev** [comment]
David's tooltips:
- https://replay.io
- https://zagjs.com/
